### PR TITLE
drivers/dht: added support for SAUL

### PR DESCRIPTION
**Pull request by haukepetersen**
~~based on #4664~~

This PR adds SAUL support for DHTxx devices.

The interesting part here is the synchronization of values for the temperature and humidity endpoints.

### CODE REVIEW

**Comment by DipSwitch** [comment]
Waiting for what PR exactly? ;)

**Comment by haukepetersen** [comment]
I guess for none anymore?! Will rebase

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
Somebody cares to test/ack, @PeterKietzmann, @LudwigKnuepfer?

**Comment by PeterKietzmann** [comment]
I will (tomorrow). Would be nice if someone could have a look at the code cause I still didn't learn about SAUL :/ ...

File: drivers/include/saul.h [hunk 1]
```diff
@@ -140,6 +140,11 @@ typedef struct {
 } saul_driver_t;
 
 /**
+ * @brief   Default not supported function
+ */
+int saul_notsup(void *dev, phydat_t *dat);
+
+/**
```
Comment (PeterKietzmann) [inline drivers/include/saul.h]: Thinking about a separate PR...

**Comment by PeterKietzmann** [comment]
Code looks fine from what I can say and it works. Question is if we should move the general SAUL commit into another PR. I personally don't care much. ACK once this is decided.

File: drivers/include/saul.h (hunk 1 above)
Comment (haukepetersen) [inline drivers/include/saul.h]: It is a separate commit, so no need for a separate PR...

**Comment by haukepetersen** [comment]
I decided it: it's a minor change and it is in a separate commit, so this should be fine. ACK is there, CI is happy -> go.

**Pull request merged**